core/sql/optimizer/NATable.cpp

```diff
@@ -5481,7 +5481,9 @@ NABoolean createNAFileSets(TrafDesc * table_desc       /*IN*/,
 
   if(postCreateNATableWarnings != preCreateNATableWarnings)
 	  tableConstructionHadWarnings_=TRUE;
-  const char *lobHdfsServer = CmpCommon::getDefaultString(LOB_HDFS_SERVER);
+  char  lobHdfsServer[256] ; // max length determined by dfs.namenode.fs-limits.max-component-length(255)
+  memset(lobHdfsServer,0,256);
+  strncpy(lobHdfsServer,CmpCommon::getDefaultString(LOB_HDFS_SERVER), strlen(CmpCommon::getDefaultString(LOB_HDFS_SERVER)));
```

Comment:
The comment in the next file tells us that there is expectation the max valid string length will be 255 bytes for this attribute. However with user error there could be a longer invalid string. Maybe some error handling code that checks if the specifies string is less that 255 bytes will address the concern above.

I do have another question.
We are passing a pointer to a local variable (lobHdfsSever) to the CLI call SQL_EXEC_LobddlInterface(). Is there a guarantee that this call will NOT save the pointer somewhere that could cause the pointer to be accessed after the call has returned. I suppose I am not clear on changing the storage of this actual data from system heap used by NAString (old code) to stack used by newly created char array will avoid the corruption problem. I am often confused by such question, so consider this a question, rather than a issue to be resolved.